From 239fa64ffd743613e759403950a4740808fd8d3b Mon Sep 17 00:00:00 2001
From: Han Xu <b45815@freescale.com>
Date: Thu, 4 Jun 2015 11:48:23 -0500
Subject: [PATCH 0231/5242] MLK-11044: mtd:nand: enable per1_bch for NAND on
 all i.MX6 platform

commit  a4a136493a41be3392936bb839ccba738b4f7a50 from
https://source.codeaurora.org/external/imx/linux-imx.git

The per1_bch was moved in patch below since it was never mentioned in
any GPMI/BCH/APBH documents, but actually it is necessary for BCH module
since BCH use AXI bus transfer data through fabric, need to enable this
clock for BCH at fabric side.

This patch enabled this clock for all i.MX6 platforms and has been
tested on i.MX6Q/i.MX6QP/i.MX6SX and i.MX6UL.

commit 9aa0fb0a606a583e2b6e19892ac2cab1b0e726c4
Author: Han Xu <b45815@freescale.com>
Date: Thu May 28 16:49:18 2015 -0500

mtd: nand: support NAND on i.MX6UL

support i.MX6UL GPMI NAND driver and removed the unecessary clock
per1_bch.

Signed-off-by: Han Xu <b45815@freescale.com>

Fixed in 4.14 rebase for different clock array.

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c |    2 +-
 drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.h |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
index a168b7b..5f40e49 100644
--- a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
+++ b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
@@ -108,7 +108,7 @@ static int gpmi_ooblayout_free(struct mtd_info *mtd, int section,
 };
 
 static const char * const gpmi_clks_for_mx6[] = {
-	"gpmi_io", "gpmi_apb", "gpmi_bch", "gpmi_bch_apb",
+	"gpmi_io", "gpmi_apb", "gpmi_bch", "gpmi_bch_apb", "per1_bch",
 };
 
 static const struct gpmi_devdata gpmi_devdata_imx6q = {
diff --git a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.h b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.h
index 87dbc3a..442b358 100644
--- a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.h
+++ b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.h
@@ -22,7 +22,7 @@
 #include <linux/dma-mapping.h>
 #include <linux/dmaengine.h>
 
-#define GPMI_CLK_MAX 4 /* MX6Q needs four clocks */
+#define GPMI_CLK_MAX 5 /* MX6Q needs five clocks */
 struct resources {
 	void __iomem  *gpmi_regs;
 	void __iomem  *bch_regs;
-- 
1.7.9.5

